src/Tokenizer/Analyzer/BlocksAnalyzer.php

```diff
@@ -24,18 +24,14 @@
  */
 final class BlocksAnalyzer
 {
-    public function isBlock(Tokens $tokens, ?int $openIndex, ?int $closeIndex): bool
+    public function isBlock(Tokens $tokens, int $openIndex, int $closeIndex): bool
     {
-        if (null === $openIndex || null === $closeIndex) {
-            return false;
-        }
-
         if (!$tokens->offsetExists($openIndex)) {
-            return false;
+            throw new \InvalidArgumentException(sprintf('Token open index %d does not exist.', $openIndex));
```

Comment:
Shouldn't it be like this?

```suggestion
            throw new \InvalidArgumentException(sprintf('Block open index %d does not exist.', $openIndex));
```

"Token open index" looks weird 🤔.